### ISSUE TITLE
Don't use OpenSSL but Apple's Security services for signature verification

### ISSUE THREAD

**Issue by arianvp** [minvws/nl-covid19-notification-app-ios#22]
### Describe the bug, issue or concern

Vendoring OpenSSL puts burden on keeping it up to date manually; which is error prone.

It currently is used for  verifying   PKCS#7/CMS detached signatures as defined https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/architecture/Crypto%20Raamwerk.md ;    however  Apple's Security services already has support for this built-in directly:

https://developer.apple.com/documentation/security/cryptographic_message_syntax_services

I would suggest using the Apple APIs directly instead of vendoring OpenSSL to minimize dependency footprint and attack surface.

**Comment by rvandijke** [comment]
Hi @arianvp,

Initially the idea was to use those API's. I even created a working playground, until I found out the APIs are only available on OSX (and not on iOS). Unfortunately CMSDecoder and related APIs are not an option and that's why I had to fallback on using OpenSSL.

**Comment by arianvp** [comment]
oooh that's a shame!  That seems a very valid reason though.  Totally understand

**Comment by arianvp** [comment]
I guess I will close this issue then. Thanks for the context.

**Comment by bbaassssiiee** [comment]
Secura's audit found out you didn't update to the latest OpenSSL. This should be part of continuous integration.

**Comment by rvandijke** [comment]
The OpenSSL library was updated to version 1.1.1g about two weeks ago so the issue in the report is resolved now. We should’ve used that version from the start, but it wasn’t configured in the iPhone-for-OpenSSL build script we used. My bad for not looking into this enough. 

I don’t necessarily agree this should be part of CI: when upgrading third party libraries you want to make sure everything works properly and perform several tests to ensure the update works correctly with your implementation. I’ve had situations in the past where updating to new third party software caused crashes, memory leaks or performance issues.
